cmd/exec.go

```diff
@@ -56,25 +56,44 @@ var execCmd = &cobra.Command{
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 
+		if name == "" {
+			name = c.Config.Name
+		}
+
+		filters := types.FilterFromLabelStrings(labelsFilter)
+
+		// list all nodes matching the filters
+		nodes, err := c.ListNodes(ctx, filters)
+		if err != nil {
+			return err
+		}
+
+		// prepare the exec collection and the exec command
 		resultCollection := exec.NewExecCollection()
 
-		for _, node := range c.Nodes {
-			for _, execCommand := range execCommands {
-				execCmd, err := exec.NewExecCmdFromString(execCommand)
-				if err != nil {
-					// do not stop exec for other nodes if some failed
-					log.Error(err)
-				}
+		// build execs from the string input
+		var execCmds []*exec.ExecCmd
+		for _, execCmdStr := range execCommands {
+			execCmd, err := exec.NewExecCmdFromString(execCmdStr)
+			if err != nil {
+				return err
+			}
+			execCmds = append(execCmds, execCmd)
+		}
 
+		// run the exec commands on all the cotnainers matching the filter
+		for _, node := range nodes {
+			// iterate over the commands
+			for _, execCmd := range execCmds {
+				// execute the commands
 				execResult, err := node.RunExec(ctx, execCmd)
 				if err != nil {
 					// skip nodes that do not support exec
 					if err == exec.ErrRunExecNotSupported {
 						continue
 					}
-					return err
 				}
-				resultCollection.Add(node.Config().ShortName, execResult)
+				resultCollection.Add(node.Cfg.ShortName, execResult)
```

Comment:
Hi @steiler 
I propose we put a requirement for users to always specify a topology file as an argument for `clab exec` command. That way we are able to init CLab structure and init all the node.

With that we are able to add labels to nodes which are not orchestrated by clab (i.e. ext-container nodes).

To make filters to work, we have to implement `c.ListNode(ctx, filters)` function that should return a slice of clab Nodes interfaces which satisfy to the provided filter.

Then we can leverage nodes RunExec function without the need to have execs implemented for generic containers, as we will work only with nodes